components/vfs/manager.cpp

```diff
@@ -107,4 +108,31 @@ namespace VFS
         normalize_path(name, mStrict);
     }
 
+    const std::string& Manager::findFirstOf(const std::string &name) const
+    {
+        std::string normalized = name;
+        normalize_path(normalized, mStrict);
+        size_t extensionPos = normalized.rfind('.');
+        if (extensionPos == std::string::npos || normalized.find('/', extensionPos+1) != std::string::npos)
+            extensionPos = normalized.length();
+        const std::string basename = normalized.substr(0, extensionPos);
+
+        // Search starting from archives with higher priority
+        for (auto iter = mArchiveIndexes.rbegin(); iter != mArchiveIndexes.rend(); ++iter)
+        {
+            NameFileMap::const_iterator entry = iter->lower_bound(basename);
+            if (entry != iter->end() && entry->first.length() >= basename.length())
+            {
+                extensionPos = entry->first.rfind('.');
+
+                if (extensionPos == std::string::npos || entry->first.find('/', extensionPos+1) != std::string::npos)
+                    extensionPos = entry->first.length();
+                if (entry->first.compare(0, extensionPos, basename) == 0)
+                    return entry->first;
+            }
+        }
+        
+        return {};
```

Comment:
I'm not sure this is valid, since this constructs a temporary which is then bound to and returned as a const reference. The reference becomes invalid almost immediately after returning. You can instead do:
```
static const std::string empty;
return empty;
```

Reply:
>I'm not sure this is valid, since this constructs a temporary which is then bound to and returned as a const reference. 

Thing is, findFirstOf currently doesn't return a const reference, otherwise the game'd crash at launch if the requested file doesn't exist and a reference to freshly purged empty string was passed over.

Reply:
Right, I noticed you changed that in a later commit after I wrote the comment.